examples/knative/python/kn-py-vm-attr/vcenter.py

```diff
@@ -0,0 +1,111 @@
+import os
+import atexit
+import ssl
+import logging
+from pyVim.connect import SmartConnect, Disconnect
+from pyVmomi import vim
+from dotenv import load_dotenv
+
+logger = logging.getLogger(__name__)
+
+class Session:
+    """Session is a helper object to connect to a vcenter server and set custom attributes to
+       VM objects.
+    """
+
+    def __init__(self):
+        """Session is a helper object to connect to a vcenter server and set custom attributes to
+           VM objects.
+        """
+        load_dotenv() # take environment variables from .env.
+        self.ssl_context = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
+        if os.environ.get('VC_SSLVERIFY', True) == 'False':
+            self.ssl_context.verify_mode = ssl.CERT_NONE
+        try:
+            self.host = os.environ['VC_SERVER']
+            self.user = os.environ['VC_USER']
+            self.pwd  = os.environ['VC_PASSWORD']
+            self.attr_owner = os.environ['VC_ATTR_OWNER']
+            self.attr_creation_date = os.environ['VC_ATTR_CREATION_DATE']
+            self.attr_last_poweredon = os.environ['VC_ATTR_LAST_POWEREDON']
+        except KeyError as err:
+            raise Exception(f'Missing mandatory configuration key: {err}')
+
+        logger.debug(f'Initializing vCenter connection...')
+        try:
+            self.service_instance = SmartConnect(
+                host = self.host,
+                user = self.user,
+                pwd  = self.pwd,
+                port = 443,
+                sslContext = self.ssl_context
+            )
+            atexit.register(Disconnect, self.service_instance)
+            self.content = self.service_instance.RetrieveContent()
+        except IOError as err:
+            raise Exception(f'Error connecting to vCenter: {err}')
+        except Exception as err:
+            raise Exception(f'Unknown error when creating vsphere session: {err}')
+        logger.info(f"Connected to vCenter {self.host}")
+
+
+    def close(self):
+        """Disconnect the current session from vCenter.
+        """
+        Disconnect(self.service_instance)
+
+
+    def get_vm_attributes(self):
```

Comment:
I found this method name a bit misleading as it's getting attributes, not neccessarily tied to a VM...would `get_field_attributes` be a better naming?